.github/workflows/full_kubeflow_integration_test.yaml

```diff
@@ -4,278 +4,178 @@ on:
+        ./tests/gh-actions/install_multi_tenancy.sh
         kustomize build ./common/dex/overlays/oauth2-proxy | kubectl apply -f -
 
-        echo "Waiting for pods in auth namespace to become ready..."
-        kubectl wait --for=condition=Ready pods --all --timeout=180s -n auth
-    - name: Install central-dashboard
+    - name: Install Central Dashboard and Pipelines
       run: |
         kustomize build apps/centraldashboard/upstream/overlays/kserve | kubectl apply -f -
+        ./tests/gh-actions/install_pipelines.sh
         kubectl wait --for=condition=Ready pods --all --all-namespaces --timeout 180s
-    - name: Create Kubeflow User Profile
-      run: kustomize build common/user-namespace/base | kubectl apply -f -
-    - name: Verify User Profile Setup
+
+    - name: Create User Profile
       run: |
-        # Wait for profile controller to process the request
+        kustomize build common/user-namespace/base | kubectl apply -f -
         sleep 60
-
-        # Verify profile resources are properly created
-        KF_PROFILE=kubeflow-user-example-com
-        kubectl -n $KF_PROFILE get pods,configmaps,secrets
-
-        # Verify minio secret exists (critical for ML pipelines)
         if ! kubectl get secret mlpipeline-minio-artifact -n $KF_PROFILE > /dev/null 2>&1; then
-            echo "Error: Secret mlpipeline-minio-artifact not found in namespace $KF_PROFILE"
             exit 1
         fi
-    - name: Set up Python Environment
-      uses: actions/setup-python@v4
-      with:
-        python-version: '3.12'
-    - name: Install Python Dependencies
+
+    - name: Install Jupyter and Notebook Components
       run: |
+        kustomize build apps/jupyter/jupyter-web-app/upstream/overlays/istio/ | kubectl apply -f -
+        kustomize build apps/jupyter/notebook-controller/upstream/overlays/kubeflow/ | kubectl apply -f -
+        kustomize build apps/admission-webhook/upstream/overlays/cert-manager | kubectl apply -f -
+        kubectl get crd poddefaults.kubeflow.org || kubectl apply -f https://raw.githubusercontent.com/kubeflow/kubeflow/master/components/admission-webhook/manifests/base/crd.yaml
+        kubectl wait --for=condition=Ready pods --all --all-namespaces --timeout 300s --field-selector=status.phase!=Succeeded
+
+    - name: Install ML Components
+      run: |
+        # Katib
+        sudo apt-get install -y apparmor-profiles
+        sudo apparmor_parser -R /etc/apparmor.d/usr.sbin.mysqld
+        cd apps/katib/upstream && kustomize build installs/katib-with-kubeflow | kubectl apply -f - && cd ../../../
+        kubectl wait --for=condition=Available deployment/katib-controller -n kubeflow --timeout=300s
+        kubectl wait --for=condition=Available deployment/katib-mysql -n kubeflow --timeout=300s
+        kubectl label namespace $KF_PROFILE katib.kubeflow.org/metrics-collector-injection=enabled --overwrite
+
+        # Other ML components
```

Comment:
No one step per component. One step to install one component. No "Other ML Components"

Reply:
![image](https://github.com/user-attachments/assets/1f007869-a01e-4752-844e-f87392deb97d)
 they all must be consistent and rely on the reusable scripts.